docs/book/src/topics/wasi.md

```diff
@@ -0,0 +1,117 @@
+# WebAssembly / WASI Workloads
+
+## Overview
+
+CAPZ enables you to create WebAssembly (Wasm) / WASI pod workloads targeting either [Deislabs Slight](https://github.com/deislabs/spiderlightning) or [Fermyon Spin](https://github.com/fermyon/spin) frameworks for building and running fast, secure microservices with Wasm on Kubernetes (v1.23.16+, v1.24.10+, v1.25.6+, v1.26.1+, and newer Kubernetes versions).
+
+Both of the runtimes (slight and spin) for running Wasm workloads use [Wasmtime](https://wasmtime.dev) embedded in containerd shims via the [deislabs/containerd-wasm-shims](https://github.com/deislabs/containerd-wasm-shims) project which is built upon [containerd/runwasi](https://github.com/containerd/runwasi). These containerd shims enable Kubernetes to run Wasm workloads without needing to embed the Wasm runtime in each OCI image. 
+
+## Slight (SpiderLightning)
+Slight (or Spiderlightning) is an open source wasmtime-based runtime that provide cloud capabilities to Wasm microservices. These capabilities include key/value, pub/sub, and much more.
+
+## Fermyon Spin
+"Spin is an open source framework for building and running fast, secure, and composable cloud microservices with WebAssembly. It aims to be the easiest way to get started with WebAssembly microservices, and takes advantage of the latest developments in the WebAssembly component model and Wasmtime runtime."
+
+### Applying the Wasm Runtime Classes
+By default, CAPZ virtual machines include containerd shims to run both `slight` and `spin` workloads. To inform Kubernetes about the ability to run these workloads on CAPZ nodes, you will need to apply a runtime class for each runtime.
```

Comment:
```suggestion
By default, CAPZ reference virtual machine images include containerd shims to run both `slight` and `spin` workloads. To inform Kubernetes about the ability to run these workloads on CAPZ nodes, you will need to apply a runtime class for each runtime.
```

Reply:
Actually, is there a minimum image-builder / k8s version for which our images have this? My first attempt failed with

> Failed to create pod sandbox: rpc error: code = Unknown desc = failed to get sandbox runtime: no runtime for "spin" is configured

I'm using k8s v1.26.0

Reply:
looks like https://github.com/kubernetes-sigs/image-builder/pull/1037 went in Dec 22nd and v1.26.0 was published Dec 12th so I might need a newer image (I wouldn't expect a user to know how to find this out easily so worth documenting IMO)

Reply:
I believe I call out the K8s versions in the overview section. Is that sufficient?

Reply:
Per your error, that is expected given there is no runtime configured for spin in the containerd config.

Reply:
I think only the Azure ref images for v1.26.1 and later have the WASM shims included.

Reply:
@mboersma why would they not have been included in say v1.23.16 and others that were published at the same time as 1.26.1?

Reply:
@devigned, sorry, I misspoke. I was thinking for Cecile that she needs v1.26.1, but all the recent set of patches (v1.23.16 and friends) have the Wasm shims.

Reply:
> I believe I call out the K8s versions in the overview section. Is that sufficient?

Ah I somehow missed that (or rather read it then forgot about it by the time I got the next step 😬 . Should be good enough!